lib/elixir/lib/macro.ex

```diff
@@ -2383,7 +2383,7 @@ defmodule Macro do
     end
   end
 
-  defp valid_alias?('Elixir' ++ rest), do: valid_alias_piece?(rest)
+  defp valid_alias?([?E, ?l, ?i, ?x, ?i, ?r] ++ rest), do: valid_alias_piece?(rest)
```

Comment:
```suggestion
  defp valid_alias?([?E, ?l, ?i, ?x, ?i, ?r, ?.] ++ rest), do: valid_alias_piece?(rest)
```

Reply:
The original list was `'Elixir'` here, not `'Elixir.'` (the dot is checked in `valid_alias_piece?`)